### PR TITLE
add support for push notification

### PR DESCRIPTION
**Pull request by hankim813**
Ticket reference: https://segment.zendesk.com/agent/tickets/31938

Mixpanel Docs: https://mixpanel.com/help/reference/ios-push-notifications
Our Specs for sending device tokens: https://segment.com/docs/spec/common/#context

Users have to manually pass the device token in their `.identify()` call via `context.device.token` and then we will send that to mixpanel's /engage endpoint. Must batch the requests separately for `$set` and `$union`.

Also made a minor naming change in track to follow naming convention -- `callback` to `fn` in `batch.end()`;

@ndhoule @sperand-io @f2prateek

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -75,10 +78,35 @@ Mixpanel.prototype.identify = function(identify, fn){
     data: b64encode(payload)
```
Comment (ndhoule) [inline lib/index.js]: Move `query` inside the `batch.push` call; data should be as local as possible, and we only use it in there

File: lib/index.js [hunk 2]
```diff
@@ -75,10 +78,35 @@ Mixpanel.prototype.identify = function(identify, fn){
...
+      .query(query)
+      .end(self._parseResponse(fn));
+  });
+
+  // https://mixpanel.com/help/reference/ios-push-notifications
+  // Send device token for iOS for push notification support
+  var deviceToken = identify.proxy('context.device.token');
+
+  if (deviceToken) {
+    delete payload.$set;
```
Comment (ndhoule) [inline lib/index.js]: This is pretty brittle; make a base `payload` and extend call-specific objects instead. E.g.:

```js
var basePayload = {
  // ...
};

batch.push(function(fn) {
  var payload = extend({
    $set: // ...
  }, basePayload);
  self
    .get('/engage')
    .query(b64encode(payload))
    .end(self._parseResponse(fn));
});
```

Comment (hankim813) [inline lib/index.js]: also do we need `batch.throws(true)`?

**Comment by ndhoule** [comment]
So long as `$set` and `$union` can be called in any order (batch can't guarantee requests will complete in order) then lgtm. @hankim813 were you able to test that out?

**Pull request merged**